### PR TITLE
Fix/string expand fields on leaves

### PR DESCRIPTION
**Pull request by CostantiniMatteo**
Hi guys! I'm opening this PR this fix what I think it's a bug.

When calling `JSONAPI.Utils.String.expand_fields`, the current behaviour is to always apply the transformation function to every element of a list. However, when the list is the value of a map, such strings are actual values and should be left as-is.

For example:
```
> expand_fields(%{"attributes" => %{"corgiName" => ["Wardel"]}}, &underscore/1)
%{"attributes" => %{"corgi_name" => ["wardel"]}}
```

Here `Wardel` is not a field which should be recased, but an actual value that should not change.

If the current implementation is expected behaviour, feel free to close this PR.

Thanks!

### CODE REVIEW

**Comment by CostantiniMatteo** [comment]
I just saw that #239 solves the same problem. The difference between the two is that in this PR objects in lists are transformed, while in the other they left as-is.

File: .github/workflows/ci.yml
Comment (doomspork) [inline .github/workflows/ci.yml:24]: What does this change get us? I'm not sure this is in scope for the propose PR.

**Review by doomspork — COMMENTED** [review]
Thanks for the PR @CostantiniMatteo. I'm not sure how #245 slipped through my notifications. I'm going to ping the other maintainers so we can get eyes on these 2 and decide which to merge.

File: .github/workflows/ci.yml
Comment (jherdman) [reply .github/workflows/ci.yml:24]: The change is for a non-deprecated action to set up Elixir with Github Actions. It's a good change, though I'm not sure it should be part of this PR.

**Review by jherdman — COMMENTED** [review]
IMHO the simpler PR is better, and we update CI in a different PR.

File: .github/workflows/ci.yml
Comment (CostantiniMatteo) [reply .github/workflows/ci.yml:24]: You’re right, I can open a new PR with just this change if you like.

**Comment by CostantiniMatteo** [comment]
> IMHO the simpler PR is better, and we update CI in a different PR.

In general I agree on choosing the simpler one. Just note  however that, while both PR solve the same problem, sometimes results are different (when a value is a list which contains an object). So it might also be worth taking this into account.

The following map:
```
%{someList: [“someValue”, %{someField: “someOtherValue”}]}
```

With this PR results in:
```
%{“some_list” => [“someValue”, %{“some_field” => “someOtherValue”}]}
```

And in this other map with #239:
```
%{“some_list” => [“someValue”, %{“someField” => “someOtherValue”}]}
```

I’m not sure about which choice is better. I didn’t think about leaving object values unchanged but it seems reasonable now.

I’m writing this on mobile, apologies for any mistake.

**Comment by opodartho** [comment]
@CostantiniMatteo As this and #239 solve same issue but changing the keys inside List seems more reasonable.

File: .github/workflows/ci.yml
Comment (doomspork) [reply .github/workflows/ci.yml:24]: @CostantiniMatteo can you pull this out into a separate PR and then update this one? I think both should be good to go.

**Comment by CostantiniMatteo** [comment]
I've opened #246 to fix the CI issues and cleaned-up this. Let me know if there's anything else I can do!

**Comment by lucacorti** [comment]
@CostantiniMatteo Can you update your PR with origin master so this is mergeable?

**Comment by lucacorti** [comment]
@jherdman @doomspork Is this ok?

**Comment by jherdman** [comment]
I believe this is OK, but I'm a bit out of the loop with this project as of late. I defer to @doomspork

**Review by doomspork — APPROVED** [review]
This looks good to me @snewcomer any thoughts before we merge this?

**Comment by lucacorti** [comment]
@jherdman @doomspork @snewcomer Thanks!

We are using the library at https://github.com/dottori-it and are currently consuming it from git since we need a few improvements made since 1.3.0.

If we can help with maintenance please let us know ❤️

**Comment by doomspork** [comment]
Always wonderful to learn about folks using these libraries 🎉 We're definitely open to help supporting the library 😁

**Pull request merged**
